<launch>
  <arg name="robot_dof" default="7" />
  <arg name="add_gripper" default="false" />
  <arg name="namespace" default="xarm"/>
  <arg name="no_gui_plan" default="true" />
  <arg name="robot_type" default="xarm"/>
  <arg name="model1300" default="false" />
  <arg name="num_trials" default="50" />
  <arg name="fixed_goal_x" default="0.3" />
  <arg name="fixed_goal_y" default="0.0" />
  <arg name="fixed_goal_z" default="0.4" />

  <!-- Include MoveIt configuration -->
  <include file="$(find xarm7_moveit_config)/launch/moveit_rviz_common.launch">
    <arg name="jnt_stat_source" value="[/$(arg namespace)/joint_states]" />
    <arg name="load_move_group" value="true" />
    <arg name="pub_tf" value="false" />
    <arg name="NO_GUI_CTRL" value="$(arg no_gui_plan)"/>
    <arg name="sim_gazebo" value="true"/>
    <arg name="add_gripper" value="$(arg add_gripper)"/>
  </include>
  
  <!-- Set robot name parameter -->
  <param name="robot_name" type="str" value="$(eval arg('robot_type') if arg('robot_type') == 'uf850' else arg('robot_type') + str(arg('robot_dof')))"/>
  
  <!-- Set testing parameters -->
  <param name="num_trials" value="$(arg num_trials)" />
  <param name="fixed_goal_x" value="$(arg fixed_goal_x)" />
  <param name="fixed_goal_y" value="$(arg fixed_goal_y)" />
  <param name="fixed_goal_z" value="$(arg fixed_goal_z)" />
  
  <!-- Launch the planner tester -->
  <node name="planner_tester" pkg="vader_planner" type="planner_tester" output="screen">
    <param name="num_trials" value="$(arg num_trials)" />
    <param name="fixed_goal_x" value="$(arg fixed_goal_x)" />
    <param name="fixed_goal_y" value="$(arg fixed_goal_y)" />
    <param name="fixed_goal_z" value="$(arg fixed_goal_z)" />
  </node>
  
  <!-- Optional: Launch RViz for visualization -->
  <group if="$(eval not arg('no_gui_plan'))">
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find vader_planner)/launch/planner.rviz" />
  </group>
  
</launch> 